File: KITE-Simulcast-Test/js/steps/LoadPageStep.js
const {TestStep} = require('kite-common');

/**
 * Class: LoadPageStep
 * Extends: TestStep
 * Description:
 */
class LoadPageStep extends TestStep {
  constructor(kiteBaseTest) {
    super();
    this.driver = kiteBaseTest.driver;
    this.timeout = kiteBaseTest.timeout;
    this.url = kiteBaseTest.url;
    this.page = kiteBaseTest.page;
  }

  stepDescription() {
    return 'Open ' + this.url;
  }

  async step() {
    await this.page.open(this);
  }
}

module.exports = LoadPageStep;

File: KITE-Simulcast-Test/src/main/java/io/cosmosoftware/kite/simulcast/steps/LoadPageStep.java
package io.cosmosoftware.kite.simulcast.steps;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebDriver;

import static io.cosmosoftware.kite.entities.Timeouts.ONE_SECOND_INTERVAL;
import static io.cosmosoftware.kite.util.TestUtils.waitAround;
import static io.cosmosoftware.kite.util.WebDriverUtils.loadPage;

public class LoadPageStep extends TestStep {

  private final String url;

  
  public LoadPageStep(Runner runner, String url) {
    super(runner);
    this.url = url;
  }
  
  @Override
  public String stepDescription() {
    return "Open " + url;
  }
  
  @Override
  protected void step() throws KiteTestException {
    loadPage(webDriver, url, 20);
    waitAround(ONE_SECOND_INTERVAL);
  }
}
